packages/manager/apps/dedicated/client/app/account/identity-documents/user-identity-documents.controller.js

```diff
@@ -72,35 +90,53 @@ export default class AccountUserIdentityDocumentsController {
   }
 
   getUploadDocumentsLinks(count) {
-    return this.$http
-      .post(`/me/procedure/identity`, {
-        numberOfDocuments: count,
-      })
-      .then(({ data: response }) => {
-        const { uploadLinks } = response;
-        return this.$q.all(
-          uploadLinks.map((uploadLink, index) =>
-            this.uploadDocumentsToS3usingLinks(uploadLink, this.files[index]),
-          ),
-        );
-      })
-      .then(() => {
-        this.$http.post(`/me/procedure/identity/finalize`);
-        this.showUploadOption = false;
-      })
-      .catch(() => {
-        this.displayErrorBanner();
-      });
+    return this.$http.post(`/me/procedure/identity`, {
+      numberOfDocuments: count,
+    });
   }
 
-  uploadDocumentsToS3usingLinks(uploadLink, uploadedfile) {
-    return this.$http
-      .put(uploadLink.link, uploadedfile, {
-        headers: { ...uploadLink.headers },
-      })
-      .catch(() => {
-        this.displayErrorBanner();
-      });
+  // In order to avoid false positive from file upload endpoint, we'll retry all the process from the start
+  tryToFinalizeProcedure(uploadLinks) {
+    let remainingRetries = MAX_RETRIES;
+    // First we try to upload the file using the retrieved link
+    return (
+      this.finalizeProcedure(uploadLinks)
+        // In case of any error, we'll retry until we hit some maximum retry amount
+        .catch(async () => {
+          let success = false;
+          // As we need to retry until the task in is success, or we exceed the max retry count, the following await
+          // in a loop is valid, so we'll ignore eslint on the following loop
+          while (!success && remainingRetries) {
+            /* eslint-disable no-await-in-loop */
+            await new Promise((resolve) => {
+              setTimeout(resolve, DELAY_BETWEEN_RETRY);
+            });
+            try {
+              /* eslint-disable no-await-in-loop */
+              await this.finalizeProcedure(uploadLinks);
+              success = true;
+            } catch (error) {
+              remainingRetries -= 1;
+            }
+          }
+          return success ? this.$q.resolve() : this.$q.reject();
+        })
```

Comment:
**question:** Does not this have any side-effect with respect to digest-cycle. Because AngularJS does not pick changes in `setTimeout` and `async await` and update the UI which would require to run the digest cycle manually.

Reply:
It should not as in this function we do not make changes on UI, the UI changes are done in `then` or `catch` triggered by this Promise